demisto_sdk/commands/coverage_analyze/tests/coverage_report_test.py

```diff
@@ -0,0 +1,195 @@
+        monkeypatch.chdir(tmpdir)
+        assert CoverageReport().original_summary == json.loads(read_file(JSON_MIN_DATA_FILE))['files']
+
+
+class TestFileMinCoverage:
+
+    data_test_with_new_file = [
+        ('test', 70.0),
+        ('/Users/username/dev/content/Packs/SomePack/Integrations/SomeIntegration/SomeIntegration.py', 80.0),
+    ]
+
+    @pytest.mark.parametrize('file_path, default_min_cover', data_test_with_new_file)
+    def test_with_new_file(self, file_path, default_min_cover, tmpdir, monkeypatch):
+        monkeypatch.chdir(tmpdir)
+        cov_report = CoverageReport(default_min_coverage=default_min_cover)
+        cov_report._original_summary = {}
+        assert cov_report.file_min_coverage(file_path) == default_min_cover
+
+    data_test_with_exist_file = [
+        ('test', 70.0, 69.0),
+        ('/Users/username/dev/content/Packs/SomePack/Integrations/SomeIntegration/SomeIntegration.py', 80.0, 79.0),
+    ]
+
+    @pytest.mark.parametrize('file_path, current_cover, expected_min_cover', data_test_with_exist_file)
+    def test_with_exist_file(self, file_path, current_cover, expected_min_cover, tmpdir, monkeypatch):
+        file_path = os.path.relpath(file_path)
+        monkeypatch.chdir(tmpdir)
+        cov_report = CoverageReport()
+        cov_report._original_summary = {file_path: current_cover}
+        assert cov_report.file_min_coverage(file_path) == expected_min_cover
+
+    data_test_with_custom_epsilon_file = [
+        ('test', 1.0, 79.0),
+        ('/Users/username/dev/content/Packs/SomePack/Integrations/SomeIntegration/SomeIntegration.py', 3.0, 77.0),
+    ]
+
+    @pytest.mark.parametrize('file_path, epsilon, expected_min_cover', data_test_with_custom_epsilon_file)
+    def test_with_custom_epsilon_file(self, file_path, epsilon, expected_min_cover, tmpdir, monkeypatch):
+        file_path = os.path.relpath(file_path)
+        monkeypatch.chdir(tmpdir)
+        cov_report = CoverageReport(allowed_coverage_degradation_percentage=epsilon)
+        cov_report._original_summary = {file_path: 80.0}
+        assert cov_report.file_min_coverage(file_path) == expected_min_cover
+
+
+class TestCoverageDiffReport:
+
+    @staticmethod
+    def get_coverage_report_obj():
+        cov_report = CoverageReport()
+        cov_report._original_summary = json.loads(read_file(JSON_MIN_DATA_FILE))['files']
+        return cov_report
+
+    def test_without_files(self, caplog, tmpdir, monkeypatch, mocker):
+        monkeypatch.chdir(tmpdir)
+        cov_report = self.get_coverage_report_obj()
+        mocker.patch('demisto_sdk.commands.coverage_analyze.coverage_report.CoverageReport.files', return_value={})
+        with caplog.at_level(logging.ERROR, logger='demisto-sdk'):
+            assert cov_report.coverage_diff_report()
+        # assert caplog.records == []
+
+    def test_with_degradated_files(self, caplog, tmpdir, monkeypatch, mocker):
+        monkeypatch.chdir(tmpdir)
```

Comment:
make sure the change doesn't consist beyond the test scope (can be avoided by changing dirs using a `with` statement)

Reply:
it looks that it’s working correctly